### PR TITLE
OFI: patch OFI MTL for GNI provider

### PR DESCRIPTION
**Pull request by hppritcha**
Uncovered a problem using the GNI provider with the OFI MTL.
See https://github.com/ofiwg/libfabric/issues/6194.

Related to #8001

Signed-off-by: Howard Pritchard <hppritcha@gmail.com>

### CODE REVIEW

File: ompi/mca/mtl/ofi/mtl_ofi_component.c [hunk 1]
```diff
@@ -940,6 +940,17 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
...
     }
 
+    /**
+     * Unfortunately the attempt to implement FI_MR_SCALABLE in the GNI provider
+     * doesn't work, at least not well.  Since we're asking for the 1.5 libfabric
+     * API now, we have to tell GNI we want to use Mr. Basic.  Using FI_MR_BASIC
+     * rather than FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY to stay
+     * compatible with older libfabrics.
+     */
+    if (!strcmp(prov->fabric_attr->prov_name,"gni")) {
```
Comment (wckzhang) [inline ompi/mca/mtl/ofi/mtl_ofi_component.c:950]: Do you care about any utility providers?

Comment (wckzhang) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c:950]: The suggestion here would be to do a strncasecmp so it'll look at the leftmost provider for "gni" - future proofing for any layering that will happen on top.

Comment (hppritcha) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c:950]: good idea I'll change to strncasecmp.
I'm nervous to make changes for other providers, hence the check only for gni.

Comment (hppritcha) [reply ompi/mca/mtl/ofi/mtl_ofi_component.c]: pushed change to strncmp

**Comment by hppritcha** [comment]
@wckzhang updated PR with your suggested change

**Pull request merged**